### PR TITLE
Internal: Editor - Allow in-place editing [ED-7272]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/documents-manager.js [hunk 1]
```diff
@@ -39,19 +44,43 @@ export default class extends elementorModules.ViewModule {
 		this.documentClasses[ documentType ] = documentClass;
 	}
 
-	attachDocumentsClasses() {
-		this.elements.$documents.each( ( index, document ) => this.attachDocumentClass( jQuery( document ) ) );
+	attachDocumentsClasses( $parentEl = jQuery( 'body' ) ) {
+		let documentElements = $parentEl.find( this.getSettings( 'selectors' ).document )
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply assets/dev/js/frontend/documents-manager.js:48]: this.elements.$documents changed over time, it saves the documents from the first render, but a widget can be changed and documents can be created over time.

**Review by matipojo — CHANGES_REQUESTED** [review]
Please cover with tests

File: assets/dev/js/editor/elements/views/widget.js [hunk 2]
```diff
@@ -145,6 +145,10 @@ const WidgetView = BaseWidget.extend( {
 			}, 200 );
 			// Is element empty?
 		} );
+
+		if ( self.model.get( 'hasTemplate' ) && 'remote' === self.getTemplateType() ) {
+			elementorFrontend.documentsManager.attachDocumentsClasses();
+		}
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply assets/dev/js/editor/elements/views/widget.js:151]: So here I will trigger an event?

**Comment by Nevoss** [comment]
Changed the way we approach it.

**Pull request closed without merging**